src/action.c

```diff
@@ -293,6 +306,48 @@ actions_contain_toggle_keybinds(struct wl_list *action_list)
 	return false;
 }
 
+/* Checks for *required* arguments */
+bool
+action_is_valid(struct action *action)
+{
+	char *arg_name = NULL;
+	switch (action->type) {
+	case ACTION_TYPE_EXECUTE:
+		arg_name = xstrdup("command");
```

Comment:
There is no need to heap-allocate here.
Within a frame I think it's simpler to just do `arg_name="command";` or whatever - or use a helper like `get_arg_name()` as suggested above.

Reply:
Will look into it when I'm back.